devel/delete-from-ghcr

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+#
+# Delete Nextstrain images with a specific tag from GitHub Container registry.
+#
+set -euo pipefail
+
+if [[ $# -ne 1 ]]; then
+    echo "Please provide a tag." >&2
+    exit 1
+fi
+
+tag="$1"
+
+org=nextstrain
+
+for container in base base-builder
+do
+    echo "Finding version for $org/$container:$tag ..."
+
+    # Each tag should only correspond to one package version.
+    # Pushing an existing tag will untag the existing version and add the tag
+    # to the newly pushed version.
+    id=$(curl -s \
+          -H "Accept: application/vnd.github+json" \
+          -H "Authorization: Bearer $GITHUB_TOKEN" \
+          "https://api.github.com/orgs/$org/packages/container/$container/versions" \
+        | jq --arg TAG $tag '[ .[] | select(.metadata.container.tags[] == $TAG ) ] | .[0].id' \
+    )
+    if [[ $id = null ]]; then
+        echo "$org/$container:$tag was not found." >&2
+        exit 1
+    fi
+
+    echo "Version for $org/$container:$tag is $id."
+
+    # Don't use curl --fail since the error messages in the response are useful.
+    # Instead, parse error messages and exit accordingly.
+    echo "Deleting the package version for $org/$container:$tag ..."
+    deleteVersionError=$(curl -s \
+        -X DELETE \
+        -H "Accept: application/vnd.github+json" \
+        -H "Authorization: Bearer $GITHUB_TOKEN" \
+        "https://api.github.com/orgs/$org/packages/container/$container/versions/$id" \
+    )
+    echo "$deleteVersionError"
+
+    if [[ "$deleteVersionError" =~ "You cannot delete the last tagged version of a package. You must delete the package instead." ]]; then
+
+        echo "Deleting the package $org/$container ..."
+
+        deletePackageError=$(curl -s \
+            -X DELETE \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer $GITHUB_TOKEN" \
+            "https://api.github.com/orgs/$org/packages/container/$container"
+        )
+        echo "$deletePackageError"
+
+        if [[ -n "$deletePackageError" ]]; then
+            echo "Could not delete $org/$container." >&2
+            exit 1
+        fi
+
+    elif [[ -n "$deleteVersionError" ]]; then
+        echo "Could not delete $org/$container:$tag." >&2
+        exit 1
+    fi
+done
```

Comment:
Re: [PR review comment](https://github.com/nextstrain/docker-base/pull/103#pullrequestreview-1157154143)

> Are we going to have ephemeral nextstrain/base images showing up under https://github.com/orgs/nextstrain/packages (or https://github.com/nextstrain/docker-base/packages) when CI is running?

Yes, but [the default visibility is private](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#pushing-container-images). Note that this visibility level is per-package, not package version (tag), so it's only useful if we want to keep all tags private.

> Is using ghcr.io for possibly-transient CI images going to make it awkward/difficult if we want to also publish the production images to ghcr.io in parallel to docker.io?

In this PR, I'm using the same image name/tag for both intermediate and destination registries. This won't work if the intermediate is the same as the destination, so either name or tag will have to change for the intermediate.

1. Changing the name: This would require some further parameterization of the `devel` scripts to specify image names, which are currently hardcoded as `nextstrain/base` and `nextstrain/base-builder`. A benefit would be that intermediate images will live in packages separate from final images, so intermediates can be kept private.
2. Changing the tag: This would require further expansion of the `copy-images` script to allow different src/dest tags, and intermediate images must be public.

I'm leaning towards (1), though I'm not worried about having public intermediate images if it comes to that.

Reply:
Nod. Ok. Let's cross that bridge when we get to it then. Thanks for the answers!